sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java

```diff
@@ -1144,12 +1144,38 @@ public long count(CaseCriteria caseCriteria) {
 		return Collections.emptyList();
 	}
 
-	private Predicate buildQuarantineDateFilter(CriteriaBuilder cb, Root<Case> caze, Date from, Date to) {
+	private Predicate buildQuarantineDateFilter(CriteriaBuilder cb, Root<Case> caze, Date fromDate, Date toDate) {
+		Predicate filter = null;
+		if (fromDate != null && toDate != null) {
+			filter = CriteriaBuilderHelper.and(
```

Comment:
I think these filters don't exactly match what's specified in the issue because they don't account for the situation where QUARANTINE_TO is filled in, but QUARANTINE_FROM isn't. I think we need an extra condition like this:

```
cb.or(
  cb.and(
    cb.isNull(caze.get(Case.QUARANTINE_TO)),
    cb.between(caze.get(Case.QUARANTINE_FROM), fromDate, toDate)),
  cb.and(
    cb.isNull(caze.get(Case.QUARANTINE_FROM)),
    cb.between(caze.get(Case.QUARANTINE_TO), fromDate, toDate)),
  cb.and(
    cb.greaterThanOrEqualTo(caze.get(Case.QUARANTINE_TO), fromDate),
    cb.lessThanOrEqualTo(caze.get(Case.QUARANTINE_FROM), toDate)));
```

This means that:
* If only QUARANTINE_FROM or QUARANTINE_TO are filled in, one of those need to be between start and end date
* If both are filled in, QUARANTINE_FROM needs to be before the end date and QUARANTINE_TO needs to be after the start date to make sure that at least one of the dates is between start and end date

If I'm not mistaken, that's semantically different from the current code, right? Or did I misunderstand something?

This probably needs to be adjusted for all three if cases then (even though the subsequent if cases are less complex because there's only one filter date)